src/query/service/src/pipelines/executor/executor_graph.rs

```diff
@@ -616,37 +628,25 @@ impl ScheduleQueue {
             }
         }
 
-        if !self.sync_queue.is_empty() {
-            while let Some(processor) = self.sync_queue.pop_front() {
-                if processor
-                    .graph
-                    .can_perform_task(executor.epoch.load(Ordering::SeqCst))
-                {
-                    context.set_task(ExecutorTask::Sync(processor));
-                    break;
-                } else {
-                    let mut tasks = VecDeque::with_capacity(1);
-                    tasks.push_back(ExecutorTask::Sync(processor));
-                    global.push_tasks(context.get_worker_id(), None, tasks);
-                }
+        let mut tasks_to_global = VecDeque::with_capacity(self.sync_queue.len());
+
+        if let Some(processor) = self.sync_queue.pop_front() {
+            if processor
+                .graph
+                .can_perform_task(executor.epoch.load(Ordering::SeqCst))
+            {
+                context.set_task(ExecutorTask::Sync(processor));
+            } else {
+                tasks_to_global.push_back(ExecutorTask::Sync(processor));
             }
         }
 
-        if !self.sync_queue.is_empty() {
-            let mut current_tasks = VecDeque::with_capacity(self.sync_queue.len());
-            let mut next_tasks = VecDeque::with_capacity(self.sync_queue.len());
-            while let Some(processor) = self.sync_queue.pop_front() {
-                if processor
-                    .graph
-                    .can_perform_task(executor.epoch.load(Ordering::SeqCst))
-                {
-                    current_tasks.push_back(ExecutorTask::Sync(processor));
-                } else {
-                    next_tasks.push_back(ExecutorTask::Sync(processor));
-                }
-            }
-            let worker_id = context.get_worker_id();
-            global.push_tasks(worker_id, Some(current_tasks), next_tasks);
```

Comment:
Note: This commit includes a minor fix unrelated to the PR's title.

Problem: Tasks were being incorrectly pushed to the `current_tasks` queue here.  It could cause a single task cost two points to be executed.

Solution: This change removes the redundant push to current_tasks, ensuring each task is cost one point. (Only occurred at `steal_task_to_context`)